edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/util/MetricsDeDuplication.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Diagnostics.Util
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+
+    public static class MetricsDeDuplication
+    {
+        public static IEnumerable<Metric> RemoveDuplicateMetrics(IEnumerable<Metric> metrics)
+        {
+            Dictionary<int, (Metric metric, bool)> previousValues = new Dictionary<int, (Metric, bool)>();
+
+            foreach (Metric newMetric in metrics)
```

Comment:
I think that there is a much simpler way to write this? Why not use a set as a filter? Lookat how c# implements distinct https://referencesource.microsoft.com/#System.Core/System/Linq/Enumerable.cs,4ab583c7d8e84d6d

Reply:
Your set would be similar but you could use a "key" instead of the raw metric (since the metric could be the same but with a different value)

Reply:
Added comments to clarify behavior